lib/customer.rb

```diff
@@ -0,0 +1,48 @@
+require "csv"
+# require_relative 'customers'
+
+
+class Customer
+
+
+  def self.all
+    all_customers = []
+    CSV.read("data/customers.csv").map do |row|
+      address = {}
+      id = row[0].to_i
+      email = row[1]
+      address[:street] = row[2]
+      address[:city] = row[3]
+      address[:state] = row[4]
+      address[:zip] = row[5]
+      new_customer = Customer.new(id, email, address)
+      all_customers << new_customer
+    end
+    return all_customers
+
```

Comment:
You don't need to manually create the array if you use `map`:

```suggestion
    return CSV.read("data/customers.csv").map do |row|
      address = {}
      id = row[0].to_i
      email = row[1]
      address[:street] = row[2]
      address[:city] = row[3]
      address[:state] = row[4]
      address[:zip] = row[5]

      Customer.new(id, email, address)
    end
```